### PR TITLE
ES6: Convert lib/page

### PR DESCRIPTION
**Pull request by gu-stav**
## What does this change?

Converts `lib/page` to ES6.

### CODE REVIEW

File: static/src/javascripts/lib/page.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    ];
+
+    Object.assign(match, {
+        date: config.webPublicationDateAsUrlPart(),
+        teams,
+        isLive: config.page.isLive,
+        pageType: pageTypes.find(type => type[1] === true),
+    });
+
+    if (Array.isArray(match.pageType)) {
```
Comment (GHaberis) [inline static/src/javascripts/lib/page.js]: Do we need this check if the default of `match.pageType' is `[null, true]`?

File: static/src/javascripts/lib/page.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        match.pageType = match.pageType[0];
+    }
+
+    return isit(
+        match.id || (match.pageType[0] && match.teams.length === 2),
```
Comment (GHaberis) [inline static/src/javascripts/lib/page.js]: In the if condition above we assign `match.pageType = match.pageType[0]`, this `means match.pageType` could be `null` if match.pageType is the default `[null, true]`. If `match.id` is falsey we'll then attempt to read property `[0]` of a value which is potentially `null`, which would throw an error I believe?

File: static/src/javascripts/lib/page.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+type noable = ?() => boolean;
+
+const isit = (
+    isTrue: ?(string | boolean),
+    yes: yesable,
+    no: noable,
+    arg: any
+): boolean => {
+    if (isTrue) {
+        return yes ? yes(arg || isTrue) : arg || !!isTrue;
```
Comment (SiAdcock) [inline static/src/javascripts/lib/page.js]: Are these ternaries necessary, given that `yes` and `no` are not nullable / optional?

File: static/src/javascripts/lib/page.js (hunk 2 above)
Comment (SiAdcock) [inline static/src/javascripts/lib/page.js]: I think if `pageTypes.find(type => type[1] === true)` matches nothing, then `match.pageType` becomes `undefined`, so `match.pageType[0]` will throw an error. But, yeah, the point remains 😄

Comment (SiAdcock) [inline static/src/javascripts/lib/page.js]: Ah, as discussed, the scenario I described is not possible, ignore me 😅

**Review by SiAdcock — APPROVED** [review]

**Pull request merged**